commit 00899eba260ff3edb62bc6f45c1860bc64fd59e0
Author: Michael Jeanson <mjeanson@efficios.com>
Date:   Mon Feb 5 15:22:39 2024 -0500

    x86/cet: fix shadow stack test scripts
    
    Some shadow stack test scripts use the '==' operator with the 'test'
    command to validate exit codes resulting in the following error:
    
      sysdeps/x86_64/tst-shstk-legacy-1e.sh: 31: test: 139: unexpected operator
    
    The '==' operator is invalid for the 'test' command, use '-eq' like the
    previous call to 'test'.
    
    Signed-off-by: Michael Jeanson <mjeanson@efficios.com>
    Reviewed-by: H.J. Lu <hjl.tools@gmail.com>
    (cherry picked from commit 155bb9d036646138348fee0ac045de601811e0c5)

diff --git a/sysdeps/x86_64/tst-shstk-legacy-1e-static.sh b/sysdeps/x86_64/tst-shstk-legacy-1e-static.sh
index 46f12337571127c6..0a9a164a3e8f4610 100755
--- a/sysdeps/x86_64/tst-shstk-legacy-1e-static.sh
+++ b/sysdeps/x86_64/tst-shstk-legacy-1e-static.sh
@@ -26,7 +26,7 @@ ${common_objpfx}elf/tst-shstk-legacy-1e-static
 status=$?
 if test $status -eq 77; then
   exit 77
-elif test $status == 139; then
+elif test $status -eq 139; then
   exit 0
 else
   exit 1
diff --git a/sysdeps/x86_64/tst-shstk-legacy-1e.sh b/sysdeps/x86_64/tst-shstk-legacy-1e.sh
index 31212453d9374013..3dec5623e41dc3c8 100755
--- a/sysdeps/x86_64/tst-shstk-legacy-1e.sh
+++ b/sysdeps/x86_64/tst-shstk-legacy-1e.sh
@@ -28,7 +28,7 @@ ${test_program_prefix} \
 status=$?
 if test $status -eq 77; then
   exit 77
-elif test $status == 139; then
+elif test $status -eq 139; then
   exit 0
 else
   exit 1
diff --git a/sysdeps/x86_64/tst-shstk-legacy-1g.sh b/sysdeps/x86_64/tst-shstk-legacy-1g.sh
index e84087068e6b7b15..249831e816f6589c 100755
--- a/sysdeps/x86_64/tst-shstk-legacy-1g.sh
+++ b/sysdeps/x86_64/tst-shstk-legacy-1g.sh
@@ -28,7 +28,7 @@ ${test_program_prefix} \
 status=$?
 if test $status -eq 77; then
   exit 77
-elif test $status == 139; then
+elif test $status -eq 139; then
   exit 0
 else
   exit 1
